src/Symfony/Component/RateLimiter/RateLimit.php

```diff
@@ -67,7 +67,7 @@ public function getLimit(): int
 
     public function wait(): void
     {
-        $delta = $this->retryAfter->format('U.u') - microtime(true);
+        $delta = (int) ((float) $this->retryAfter->format('U.u') - microtime(true));
```

Comment:
This change will make things worse I'm afraid because we're losing precision too early.